PROJECT_GUIDELINES.md

```diff
@@ -0,0 +1,173 @@
+
+#### Utils
+
+##### Unique IDs for components
+
+Many times it is necessary for components to have a `id="something"` attribute for things like `<label>` and various `aria-*` properties. To safely generate a unique id for a component but to also allow a user supplied `id` attribute to work follow the following pattern:
+
+```
+import { guid } from "../../utils/guid";
+
+@Component({
+  tag: "calcite-example",
+  styleUrl: "calcite-example.scss",
+  shadow: true
+})
+export class CalciteExample {
+
+  // ...
+
+  guid: string = `calcite-example-${guid()}`;
+
+  render() {
+    const id = this.el.id || this.guid;
+    return (
+      <Host id={id}></Host>
+    );
+  }
+
+  // ...
+}
+```
+
+This will create a unique id attribute like `id="calcite-example-51af-0941-54ae-22c14d441beb"` which should have a VERY low collision change since guid() generates IDs with `window.crypto.getRandomValues`. If a user supplies an `id` this will respect the users `id`.
+
+##### i18n
+
+Components should require as a few text translations as possible. In general lets users supply text values via slots and attributes. The lets user handle translations with their apps.
+
+If you component involves formatting numbers or dates use the [`Intl` APIs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl) for formating the display of numbers and dates in your component.
+
+To add RTL support to your components you should use the internal `getElementDir` helper to apply the `dir` attribute to your component. This means that your components `dir` attribute will always match the documents `dir`.
+
+```tsx
+import { Component, Host, Element, h} from "@stencil/core";
+import { getElementDir } from "../../utils/dom";
+
+@Component({
+  tag: "calcite-component",
+  styleUrl: "calcite-component.scss",
+  shadow: true
+})
+export class CalciteComponent {
+  @Element() el: HTMLElement;
+
+  // ...
+
+  render() {
+    const dir = getElementDir(this.el);
+
+    return (
+      <Host dir={dir}>
+        <!-- The rest of your component -->
+      </Host>
+    );
+  }
+}
+```
+
+You can then implement direction specific CSS with CSS variables:
+
+```scss
+:host {
+  --calcite-tabs-tab-margin-start: 1.25rem;
+  --calcite-tabs-tab-margin-end: 0;
+}
+
+:host([dir="rtl"]) {
+  --calcite-tabs-tab-margin-start: 0;
+  --calcite-tabs-tab-margin-end: 1.25rem;
+}
+```
+
+Your component and its child components can then use `var(--calcite-tabs-tab-margin-start)` to access their proper values based on the direction of the document.
+
+In future it will likely become necessary to provide sting translations for components. An example would be the `aria-label` for the `<calcite-modal>` close button. Initial research looks promising and we could likely implement one of these approaches and set a `lang` for each component similar to how we set `dir`.
```

Comment:
Do we need this section? It's more of an FYI and doesn't really provide guidance. That aside,

`In *the* future,`?
Typo: `sting` -> `string`